Portal/vip-api/src/main/java/fr/insalyon/creatis/vip/api/rest/RestApiBusiness.java

```diff
@@ -75,4 +83,46 @@ public ApiContext getApiContext(HttpServletRequest request, boolean isAuthentica
         }
         return new ApiContext(request, null, vipUser); // TODO is it necessary to always create a new instance ?
     }
+
+    public AuthenticationInfo authenticate(AuthenticationCredentials authCreds) throws ApiException {
+        String username = authCreds.getUsername(), password = authCreds.getPassword();
+        logger.debug("Verifying credential for " + username);
+        verifyCredentials(username, password);
+        logger.debug("Constructing authentication info for " + username);
+        AuthenticationInfo authInfo = new AuthenticationInfo();
+        String headerName = env.getProperty(CarminProperties.APIKEY_HEADER_NAME);
+        String apikey = getAnApikeyForUser(username); // the username is an email
+        authInfo.setHttpHeader(headerName);
+        authInfo.setHttpHeaderValue(apikey);
+        return authInfo;
+    }
+
+    private void verifyCredentials(String username, String password) throws ApiException {
```

Comment:
I find the name a bit misleading, since it actually signs the user in if the credentials are correct.

Reply:
I've improved that in a new commit.